File: libro/124_ordenar_3b.py
# Programa: Ordenar3b
# Descripción:
#   Este programa ordena 3 valores



# Procedimiento para ordenar dos datos 
def OrdenarDos(y, z):
    
    aux = 0
    if (y > z):
        aux = y
        y = z
        z = aux

    return (y, z)


# Procedimiento para leer un dato 
def LeerDato(indice):

    dato = int(input("¿Dato {:1d}? ".format(indice)))
    return dato


if __name__ == "__main__":

    (valorUno, valorDos, valorTres) = (0,0,0) # Valores a ordena
    tecla = 'S'                         # Tecla de opción 

    while (tecla!='N'):
        # -- Leer los datos -- #
        valorUno = LeerDato(1)
        valorDos = LeerDato(2)
        valorTres = LeerDato(3)
    
        # -- Ordenar los datos -- # 
        valorUno, valorDos = OrdenarDos(valorUno, valorDos)
        valorUno, valorTres = OrdenarDos(valorUno, valorTres)
        valorDos, valorTres = OrdenarDos(valorDos, valorTres)
    
        #-- Escribir los resultados -- #
        print("Datos Ordenados = {:5d} {:5d} {:5d}".format(valorUno, valorDos, valorTres))
    
        #-- Comprobar si se continúa -- #
        tecla = ' '
        
        while ((tecla != 'S') and (tecla != 'N')):
            tecla = input("¿Desea continuar(S/N )? ..")

File: libro/124_ordenar_3b.cpp
/* ********************************************************
* Programa: Ordenar3b
*
* Descripción:
*   Este programa ordena 3 valores
******************************************************** */

#include <stdio.h>
#include <math.h>

/** Procedimiento para ordenar dos datos */
void OrdenarDos( float & y, float & z ) {
  float aux;

  if (y > z) {
    aux = y;
    y = z;
    z = aux;
  }
}

/** Procedimiento para leer un dato */
void LeerDato( int indice, float & dato ) {
  printf( "¿Dato %1d? ", indice );
  scanf( "%f", &dato );
}

/* Programa principal */
int main() {

  float valorUno, valorDos, valorTres;  /* Valores a ordenar */
  char tecla;                                   /* Tecla de opción */

  tecla = 'S';

  while (tecla!='N'){

    /* -- Leer los datos -- */{
      LeerDato( 1 , valorUno );
      LeerDato( 2, valorDos );
      LeerDato( 3, valorTres );
    }

    /* -- Ordenar los datos -- */ {
      OrdenarDos( valorUno, valorDos ) ;
      OrdenarDos( valorUno, valorTres ) ;
      OrdenarDos( valorDos, valorTres ) ;
    }

    /*-- Escribir los resultados --*/{
      printf( "Datos Ordenados = \m" );
      printf( "%7.2f %7.2f %7.2f\n", valorUno, valorDos,valorTres);
    }

    /*-- Comprobar si se continúa --*/{
      tecla = ' ';
      printf( "\n¿Desea continuar(S/N )? .." );
      while ((tecla != 'S') && (tecla != 'N')) {
        scanf( "%c", &tecla );
      }
    }
  }
}
